.github/actions/setup-echo-server/action.yml

```diff
@@ -0,0 +1,69 @@
+name: setup-echo-server
+decription: |
+  This action sets up an echo server associating a name to it.
+  The name, e.g. `httpbin.org`, will be resolved to 127.0.0.1 by dnsmasq.
+
+inputs:
+  upstream_dns_server:
+    description: 'DNS server used by dnsmasq to resolve entries not yet cached'
+    required: false
+    default: '1.1.1.1'
+  echo_server_name:
+    description: 'A name dnsmasq will resolve to 127.0.0.1, e.g.: httpbin.org'
+    required: false
+  name_server_port:
+    description: 'Port dnsmasq will listen'
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: 'Setup deps - apt-get'
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y dnsmasq
+        sudo apt-get install -y apt-transport-https ca-certificates curl software-properties-common
+        curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+        echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+        sudo apt-get update
+        apt-cache policy docker-ce
+        sudo apt-get install -y docker-ce
+
+    - name: 'Setup deps - macos'
+      if: ${{ matrix.os == 'macos-latest' }}
+      run: brew install dnsmasq
+      shell: bash
+
+    - name: Start dnsmasq
+      shell: bash
+      run: |
+        sudo dnsmasq --no-hosts \
+                     --no-resolv \
+                     --port=${{ inputs.name_server_port }} \
```

Comment:
I ended up not removing it, but we don't need sudo since dnsmasq will listen on a non protected port.